pyccel/ast/numpyext.py

```diff
@@ -342,14 +342,16 @@ def __init__(self, arg):
         if not isinstance(arg, PyccelAstNode):
             raise TypeError('Unknown type of  %s.' % type(arg))
         super().__init__(arg)
-        self._dtype = arg.dtype
+        self._arg = PythonList(arg) if arg.rank == 0 else self._args[0]
+        self._arg = PythonInt(self._arg) if (isinstance(arg.dtype, NativeBool) or (isinstance(arg.dtype, NativeInteger) and self._arg.precision < 8)) else self._arg
```

Comment:
> self._arg.precision < 8

Are you sure this condition is correct on windows?

Reply:
i think it is fine since we check the bigger precision in all types, if PythonInt function will create an int32 object, to be honest, i can't think of falling cases, because precision in windows will not pass 8.

Reply:
It is not a failing case that I am worried about but an unnecessary cast. I don't think this causes a bug, I just think it adds an unnecessary node which will slow the code down at some point

Reply:
You seem to have done the opposite of what I am suggesting and are now casting all the time. To be clear my suggested improvement is:
`self._arg.precision < default_precision['int']`

Presumably windows doesn't truncate int64_t objects to int32_t?